src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1053,16 +1055,71 @@ public void Connect(IPAddress[] addresses, int port)
                 throw new NotSupportedException(SR.net_invalidversion);
             }
 
+// Disable CS0162: Unreachable code detected
+//
+// SuportsMultipleConnectAttempts is a constant; when false, the following lines will trigger CS0162.
+#pragma warning disable 162
             Exception lastex = null;
-            foreach (IPAddress address in addresses)
+            if (SocketPal.SupportsMultipleConnectAttempts)
```

Comment:
Having the compiler warning is a code smell: why not have this extracted within PAL?
Is there any use for any of the Windows implementations to have the `else` side of this check?